From d4463c7b29317393a2cf932e29860b99347536f9 Mon Sep 17 00:00:00 2001
From: Marsell Kukuljevic <marsell@joyent.com>
Date: Mon, 11 Feb 2019 22:00:57 +0100
Subject: [PATCH] TRITON-1213 VM resize fails when tmpfs is 0 and no tmpfs set
 in package Reviewed by: Orlando Vazquez <orlando@joyent.com> Approved by:
 Orlando Vazquez <orlando@joyent.com>

---
 lib/endpoints/vms.js | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/lib/endpoints/vms.js b/lib/endpoints/vms.js
index c239556..1fdc466 100644
--- a/lib/endpoints/vms.js
+++ b/lib/endpoints/vms.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 /*
@@ -706,6 +706,15 @@ function changeVm(req, res, next) {
         }
 
         req.log.debug({ params: params }, 'changeVm validated params');
+
+        // work around a mid-2018 to early-2019 VM.js platform bug, which causes
+        // resize jobs to fail on VMs with no tmpfs, even when the actual VM
+        // successfully resized
+        if (params.tmpfs == undefined && req.vm.tmpfs === 0) {
+            req.log.debug('changeVm set param.tmpfs to 0 since vm.tmpfs is 0');
+            params.tmpfs = 0;
+        }
+
         req.app.wfapi.createUpdateJob(req, params, function (err, juuid) {
             if (err) {
                 return next(err);
-- 
2.21.0

